File: tests/fixtures/js/examples.js
// JavaScript / TypeScript — see DESIGN per-language examples
function pushIt(arr, x) { arr.push(x); }            // input mutation -> dirty 20
function logIt(x) { console.log(x); }               // IO -> dirty 20
function addOne(x) { x += 1; return x; }            // primitive rebind -> clean 0
class Box { constructor(v){ this.v = v; }            // ctor self-mut -> clean
            set(v){ this.v = v; } }                  // set: self-mut outside ctor -> dirty 20 (class 10)
function build(n){ const a=[]; for(let i=0;i<n;i++) a.push(i); return a; }  // local owned -> clean 0

File: tests/fixtures/rust/examples.rs
// Rust — see DESIGN per-language examples
fn push_it(v: &mut Vec<i32>, x: i32) { v.push(x); }  // &mut param -> input mutation -> dirty
fn add(a: i32, b: i32) -> i32 { a + b }              // arithmetic -> clean
const fn double(x: i32) -> i32 { x * 2 }             // const fn auto-signal -> clean
struct S { n: i32 }
impl S {
    fn new(n: i32) -> Self { S { n } }               // constructor (builds value) -> clean
    fn set(&mut self, n: i32) { self.n = n; }        // &mut self setter -> dirty
    fn get(&self) -> i32 { self.n }                  // &self -> cannot mutate -> clean
}
fn log_it(x: i32) { println!("{x}"); }               // IO -> dirty
